### ISSUE TITLE
Update torchvision transforms -> transforms.v2

### ISSUE THREAD

**Issue by mrdbourke** [mrdbourke/pytorch-deep-learning#701]
`torchvision` v0.15+ brings in updated image transformations.

Supposedly these are faster/better than the originals and should be drop-in replacements.

See the guide here: https://pytorch.org/vision/main/transforms.html#v1-or-v2-which-one-should-i-use 

## TODO

- [ ] Replicate the existing notebook with the new transforms and see what changes
- [ ] Add the new notebook to the course with a note on the updates

**Comment by aagirre92** [comment]
Just for reference: Each time that `ToTensor()` is used with this `torchvision.transforms.v2` module, following pops up:

/usr/local/lib/python3.10/dist-packages/torchvision/transforms/v2/_deprecated.py:43: UserWarning: The transform `ToTensor()` is deprecated and will be removed in a future release. Instead, please use `v2.Compose([v2.ToImage(), v2.ToDtype(torch.float32, scale=True)]
  warnings.warn(`

Should we keep on using `ToTensor()`? What is the alternative? I have made the following test and it seems that output tensors are not the same:

```python
# TEST: Check transforms.ToTensor() and v2.ToImage() + v2.ToDtype(torch.float32,scale=True)
# torch version 2.1.0+cu118 and torchvision version 0.16.0+cu118
from torchvision.transforms import v2 # new
from torchvision import transforms # old

transform1 = transforms.ToTensor()

transform2 = transforms.Compose([
    v2.ToImage(),
    v2.ToDtype(torch.float32, scale=True)
])

img_path = "/content/data/pizza_steak_sushi/test/pizza/1152100.jpg"
img = Image.open(img_path)

first_transform = transform1(img)
second_transform = transform2(img)

torch.equal(first_transform,second_transform)
```
This outputs `False` 😞

**Comment by TwE3d** [comment]
Just stumbled upon this issue in my research into this exact question! 😄 
When using ToTensor or ToImage+ToDtype the values of the resulting tensors are the same. 
But the new method uses a torch.Tensor-Subclass "[Image](https://pytorch.org/vision/main/generated/torchvision.tv_tensors.Image.html)" for storing the tensor. 
Because of this torch.equal outputs False.

As it seems, both classes are mostly interchangeable with transforms etc, as described [here](https://pytorch.org/vision/main/auto_examples/transforms/plot_transforms_getting_started.html#passthrough-heuristic).

**Comment by whut-zhangwx** [comment]
It is not the reason of `<class 'torch.Tensor'>` and `<class 'torchvision.tv_tensors._image.Image'>`. Instead, it is the **scale** make them different. ToTensor() use 1/255 to convert a PIL image in the range [0, 255] to torch.float32 of shape (C x H x W) in the range [0.0, 1.0]. Howerver, `ToDtype(torch.float32, scale=True)` uses a different scale ratio, that causes them not equal. You can use this code to see how it happend:

```python
from PIL import Image
from torchvision.transforms import v2
import torch

path2img = "path/to/img"
img = Image.open(path2img).convert('RGB')

ToDtype = v2.Compose([
  v2.ToImage(),
  v2.ToDtype(torch.float32, scale=False),
])

todtype_img = ToDtype(img) / 255
totensor_img = v2.ToTensor()(img)
equal = torch.equal(todtype_img, totensor_img)
print(equal) # True
```

**Comment by whut-zhangwx** [comment]
The answer I posted above is wrong. I just come to realize it is not the problem of scale ratio. I found the two results of `ToTensor()` and `ToDtype()` only have very very minor difference in values, which could not caused by different scale ratio. I have not checked the source code of `ToDtype()`, but I guess it might be the problem of type conversion from float64 to float32. The dtype of result from scale operation in `ToDtype()` might be float64. So, there is a minor precision loss during the conversion of float64 to float32.

```python
from PIL import Image
from torchvision.transforms import v2
import torch

path2img = "path/to/img"
img = Image.open(path2img).convert('RGB')

ToTensor = v2.Compose([
  v2.ToTensor(),
])

ToDtype = v2.Compose([
  v2.ToImage(),
  v2.ToDtype(torch.float32, scale=True),
])

img_totensor = ToTensor(img)
img_todtype = ToDtype(img)

print(type(img_totensor), img_totensor.dtype)
# <class 'torch.Tensor'> torch.float32
print(type(img_todtype), img_todtype.dtype)
# <class 'torchvision.tv_tensors._image.Image'> torch.float32

print(torch.equal(img_totensor, img_todtype)) # False
print(img_totensor == img_todtype)
# tensor([[[False, False, False,  ...,  True,  True,  True],
#          [False, False, False,  ...,  True,  True, False],
#          [ True, False, False,  ...,  True,  True, False],
#          ...,
#          [ True,  True,  True,  ..., False, False, False]]])
print(img_totensor - img_todtype)
# tensor([[[-1.4901e-08, -1.4901e-08, -1.4901e-08,  ...,  0.0000e+00,  0.0000e+00],
#          [-1.4901e-08, -1.4901e-08, -1.4901e-08,  ...,  0.0000e+00, -1.4901e-08],
#          [ 0.0000e+00, -1.4901e-08, -1.4901e-08,  ...,  0.0000e+00, -1.4901e-08],
#          ...,
#          [ 0.0000e+00,  0.0000e+00,  0.0000e+00,  ..., -2.9802e-08, -2.9802e-08]]])
```
